### ISSUE TITLE
lua scripts seem to be executed twice

### ISSUE THREAD

**Issue by 1dancook** [andmarti1424/sc-im#636]
*sc-im version: 0.8.3-main and 0.8.3-dev*

## Description

Lua scripts seem to be executed twice. This is not easy to see unless you use `sc.lquery()`.

I thought at first that it was just `sc.lquery()` being called twice but it is everything.

## How to replicate

First, Put this in `query.lua` in lua folder:

```
q = sc.lquery('query: ')
sc.sc('newsheet "'..q..'"')
```

Second, in A0 do `\"@lua("query.lua",0)`

You will be asked two times for 'query' and then two new sheets will be made.

Expected result of course is that only one sheet is made.

**Comment by andmarti1424** [comment]
Yes. Its a matter of how the graph is evaluated. If you disable autocalc it would be executed only once I believe.

**Comment by 1dancook** [comment]
Okay, indeed there is relation to `autocalc` being on and it executing twice.

I am perhaps confused about what is meant by the second argument for `@lua()`

`@lua("{luascript}", {i})`

In the `:help` docs it seems that using `0` would not add this to the graph and using `1` does add it. I would then expect that even with `autocalc=1`  using `@lua("script.lua", 0)` would only execute the script one time (after entering the string) - or afterward when I press `@`.

Right now I get the same behavior using both `0` and `1`.

Perhaps I really misunderstand what that second argument is for. I have also tried to use `@lua` in a way where I do want it to autocalc when there are changes in other cells and it doesn't. I'd expect `autocalc` to work in that situation.

**Comment by Skyrex123** [comment]
Its actually execute 3 times for first time a lua script run on sc-im

you can check it by make a lua script file with bellow code (on this example script file name is l1.lua)

local a = sc.lgetnum(1,1)
a= a+ 1
sc.lsetnum(1,1,a)

then open sc-im
set cel B1 value become 1
then at cell A0 type \"@lua("l1.lua",0)
after push enter cel B1 will become 4

this mean lua script has been executed 3 times

**Comment by Skyrex123** [comment]
Its actually execute 3 times for first time a lua script run on sc-im

you can check it by make a lua script file with bellow code (on this example script file name is l1.lua)

local a = sc.lgetnum(1,1)
a= a+ 1
sc.lsetnum(1,1,a)

then open sc-im
set cel B1 value become 1
then at cell A0 type \"@lua("l1.lua",0)
after push enter cel B1 will become 4

this mean lua script has been executed 3 times


In lua script we can use global variabel for make our script only run one time.
